app/controllers/application_controller.rb

```diff
@@ -1,2 +1,12 @@
 class ApplicationController < ActionController::Base
+  before_action :require_login
+
+  private
```

Comment:
@JuanMaGiraldo could you please explain what the private does?

Reply:
The keyword private set the visibility of the methods that are defined after this. These methods are not accessible via router, unlike public methods that are exposed. also these methods are used as internal helper methods.